tests/python/pants_test/option/test_options.py

```diff
@@ -1200,6 +1202,28 @@ def parse_func(dest, fromfile):
 
     self.assert_fromfile(parse_func)
 
+  def test_fromfile_json(self):
+    def parse_func(dest, fromfile):
+      return self._parse('./pants fromfile --{}=@{}'.format(dest.replace('_', '-'), fromfile))
+
+    val = {'a': {'b': 1}, 'c': [2, 3]}
+    with temporary_file(suffix='.json', binary_mode=False) as fp:
+      json.dump(val, fp)
+      fp.close()
+      options = self._parse('./pants fromfile --{}=@{}'.format('dictvalue', fp.name))
+      self.assertEqual(val, options.for_scope('fromfile')['dictvalue'])
```

Comment:
If you move these two lines out of the `with` block, you should be able to avoid the call to `fp.close()`.

```python
with temporary_file(suffix='.json', binary_mode=False) as fp:
   json.dump(val, fp)
options = self._parse('./pants fromfile --{}=@{}'.format('dictvalue', fp.name))
self.assertEqual(val, options.for_scope('fromfile')['dictvalue'])
```

Reply:
No, because the file is deleted when the context exits. That's exactly what `temporary_file()` is for.